Store debug context in thread local variable

bikramSingh91 opened the pull request:
Summary: This change introduces a way to store thread local context which can be logged by installing signal handlers as it is accesible via a global function GetThreadDebugInfo(). Currently, only the query id is stored but it can be extended to add more information.

Differential Revision: D47570239

amitkdutta commented:
@bikramSingh91 This is awesome. Just for my learning, so we will see the query id and stack trace in Presto server core dump? That is gold information since we don't need to open the core.

bikramSingh91 commented:
> 

@amitkdutta Yes, you will see both in the log and the struct ThreadDebugInfo can be easily extended to add more relevant information

bikramSingh91 commented:
updating the PR with following changes:

 - moved test to velox_exec_test to avoid static linking of velox_exec_test_lib ballooning up the size of the test executable in process/test.
- Changes the test to assign a value to the nullptr since release builds where not inducing a segfault
- Switched to using ASSERT_DEATH that does not verify the type of signal since release builds encounter SIGILL and debug builds encounter SEGFAULT
